spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -16,132 +16,343 @@
 
 package com.intel.analytics.bigdl.dataset
 
+
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Storage, Tensor}
 import org.apache.commons.lang3.SerializationUtils
 
 import scala.reflect.ClassTag
 
 /**
  * Class that represents the features and labels of a data sample.
+ *
  * @tparam T numeric type
  */
 abstract class Sample[T: ClassTag] extends Serializable {
   /**
-   * Length of the feature.
+   * Length of the features.
```

Comment:
we only need on Sample now, right? User should not have the reason to develop their owns